core/src/streams.rs

```diff
@@ -331,6 +338,151 @@ impl<'gc> NetStream<'gc> {
         self.0.read().buffer.len()
     }
 
+    pub fn time(self) -> f64 {
+        self.0.read().stream_time
+    }
+
+    /// Queue a seek to be executed on the next frame tick.
+    ///
+    /// `offset` is in milliseconds.
+    pub fn seek(self, context: &mut UpdateContext<'_, 'gc>, offset: f64) {
```

Comment:
Can this have an extra argument `notify` (like with `fn pause`) that determines whether or not to broadcast the event, instead of relying on `context.is_action_script_3`?